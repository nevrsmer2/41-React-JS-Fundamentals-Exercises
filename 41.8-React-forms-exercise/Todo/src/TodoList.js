import React, { useState } from 'react';
import { v4 as uuid } from 'uuid';
import Todo from './Todo';
import TodoFormNew from './TodoFormNew';


const INITIAL_STATE = [
    { id: uuid(), todo: "Wash car" },
    { id: uuid(), todo: "Feed cats" },
    { id: uuid(), todo: "Check mail" }
];

const TodoList = () => {

    const [todo, setTodo] = useState(INITIAL_STATE);

    // useEffect(() => {
    //     console.log("TODO []:", todo);
    // }, [todo])

    const addTodo = (newTodo) => {
        setTodo(todo => [...todo, { id: uuid(), ...newTodo }]);
    };

    const deleteTodo = (id) => {
        const removed = todo.filter(todo => todo.id !== id)
        setTodo(removed);
    };

    return (
        <>
            <h1>To-do List</h1>
            <ul>
                < TodoFormNew addTodo={addTodo} />
                {todo.map(({ id, todo }) => <Todo key={id} id={id} todo={todo} deleteTodo={deleteTodo} />)}
            </ul>
        </>
    );

};


export default TodoList;