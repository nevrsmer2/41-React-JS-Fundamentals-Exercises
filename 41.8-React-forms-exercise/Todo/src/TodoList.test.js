import { render } from "@testing-library/react";
import React from "react";
import TodoList from "./TodoList";

// Smike Test
it("Todo component renders without crashing", function () {
    render(<TodoList />);
});

// Snapshot Test 
it("Current structure matches structure in snapshot", function () {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});


it("Has H1 in document", () => {
    const { getByText } = render(<TodoList />);
    const h1 = getByText("To-do List");
    expect(h1).toBeInTheDocument();
});

