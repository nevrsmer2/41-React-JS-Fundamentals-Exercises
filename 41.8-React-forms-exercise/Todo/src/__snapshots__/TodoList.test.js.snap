// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current structure matches structure in snapshot 1`] = `
<DocumentFragment>
  <h1>
    To-do List
  </h1>
  <ul>
    <form>
      <label
        for="todo"
      >
        New Task: 
      </label>
      <input
        id="todo"
        name="todo"
        placeholder="New Todo"
        type="text"
        value=""
      />
      <br />
      <button>
        Add Task
      </button>
    </form>
    <li>
      Task: Wash car
      <button>
        X
      </button>
    </li>
    <li>
      Task: Feed cats
      <button>
        X
      </button>
    </li>
    <li>
      Task: Check mail
      <button>
        X
      </button>
    </li>
  </ul>
</DocumentFragment>
`;
